torch/_dynamo/symbolic_convert.py

```diff
@@ -3699,6 +3699,70 @@ def MATCH_MAPPING(self, inst: Instruction) -> None:
         else:
             self.push(ConstantVariable.create(False))
 
+    def MATCH_CLASS(self, inst):
+        import builtins
+
+        from torch._dynamo.variables.builtin import BuiltinVariable
+        from torch._dynamo.variables.constant import ConstantVariable
+        from torch._dynamo.variables.tensor import TensorVariable
+        from torch._dynamo.variables.user_defined import UserDefinedClassVariable
+
+        from .exc import Unsupported
+
+        names = self.pop()
+        cls_type = self.pop()
+        subject = self.pop()
+
+        if hasattr(names, "realize"):
+            names = names.realize()
+        if hasattr(cls_type, "realize"):
+            cls_type = cls_type.realize()
+        if hasattr(subject, "realize"):
+            subject = subject.realize()
+
+        if isinstance(subject, TensorVariable) and isinstance(
+            cls_type, UserDefinedClassVariable
+        ):
+            match_result = ConstantVariable.create(False)
+        else:
+            try:
+                isinstance_var = BuiltinVariable(builtins.isinstance)
+                match_result = self.call_function(
+                    isinstance_var, [subject, cls_type], {}
+                )
+            except AssertionError:
+                match_result = ConstantVariable.create(False)
+            except Exception:
+                raise
+
+        if match_result is None:
+            self.push(subject)
+            self.push(cls_type)
+            self.push(names)
+            raise Unsupported("MATCH_CLASS: isinstance returned None")
+
+        try:
+            is_match = match_result.as_python_constant()
+        except NotImplementedError:
+            self.push(subject)
+            self.push(cls_type)
+            self.push(names)
+            raise Unsupported(
+                "MATCH_CLASS: cannot statically resolve isinstance"
+            ) from None
```

Comment:
This doesn't look correct to me. You're pushing the attributes on `NotImplementedError`